homeassistant/components/insteon_plm.py

```diff
@@ -6,115 +6,176 @@
 """
 import logging
 import asyncio
+import collections
 
 import voluptuous as vol
 
 from homeassistant.core import callback
-from homeassistant.const import (
-    CONF_PORT, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.const import (CONF_PORT, EVENT_HOMEASSISTANT_STOP)
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers import discovery
 
-REQUIREMENTS = ['insteonplm==0.7.5']
+REQUIREMENTS = ['insteonplm==0.8.2']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'insteon_plm'
 
 CONF_OVERRIDE = 'device_override'
+CONF_NEWNAMES = 'new_names'
 
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_PORT): cv.string,
-        vol.Optional(CONF_OVERRIDE, default=[]): cv.ensure_list_csv,
-    })
+        vol.Optional(CONF_OVERRIDE, default=[]): vol.All(
+            cv.ensure_list_csv, vol.Length(min=1)),
+        vol.Optional(CONF_NEWNAMES, default=['no']): cv.string
+        })
 }, extra=vol.ALLOW_EXTRA)
 
-PLM_PLATFORMS = {
-    'binary_sensor': ['binary_sensor'],
-    'light': ['light'],
-    'switch': ['switch'],
-}
-
 
 @asyncio.coroutine
 def async_setup(hass, config):
     """Set up the connection to the PLM."""
     import insteonplm
 
+    ipdb = IPDB()
+
     conf = config[DOMAIN]
     port = conf.get(CONF_PORT)
     overrides = conf.get(CONF_OVERRIDE)
+    newnames = conf.get(CONF_NEWNAMES)
```

Comment:
What is this supposed to be for? If it is to override names, that can be done via the entity registry now if the entities have the `unique_id` property. This should not be added as config option.

Reply:
The new names provide more description to make automations easier. The new names configuration option defaults to `False`. In the future we expect this to default to `True`.  We did not want to introduce this breaking change in this release. As you correctly stated, there are too many changes already in this release.